packages/twenty-server/src/workspace/workspace-sync-metadata/decorators/dynamic-field-metadata.interface.ts

```diff
@@ -0,0 +1,27 @@
+import { ComputedRelationFieldMetadataDecoratorParams } from 'src/workspace/workspace-sync-metadata/interfaces/reflect-computed-relation-field-metadata.interface';
+
+import { TypedReflect } from 'src/utils/typed-reflect';
+import { FieldMetadataType } from 'src/metadata/field-metadata/field-metadata.entity';
+
+export function ComputedRelationFieldMetadata(
+  params: ComputedRelationFieldMetadataDecoratorParams,
+): PropertyDecorator {
+  return (target: object, fieldKey: string) => {
+    const isSystem =
+      TypedReflect.getMetadata('isSystem', target, fieldKey) ?? false;
+    const gate = TypedReflect.getMetadata('gate', target, fieldKey);
+
+    TypedReflect.defineMetadata(
+      'computedRelationFieldMetadataMap',
+      {
+        type: FieldMetadataType.RELATION,
+        factory: params,
```

Comment:
I was confused with "factory" the first time. I can't really find a better name though but I think we can find something more explicit. 🤔

Reply:
Maybe computeFactory or relationFieldMetadataCreationFactory or something 😅

Reply:
I know haha, it took me some time to find something not too ugly 😅
Not sure about `computeFactory` as a factory is something that is computed, what about `paramsFactory` ?

Reply:
Ok for paramsFactory 👍 @magrinj